src/tests/Microsoft.Diagnostics.Monitoring.EventPipe/EventTracePipelineUnitTests.cs

```diff
@@ -70,6 +69,35 @@ await using (var testExecution = StartTraceeProcess("TraceStopTest"))
             Assert.True(foundCpuProvider);
         }
 
+        [Fact]
+        public async Task TestEventStreamCleanup()
+        {
+            Stream eventStream = null;
+
+            await using (var testExecution = StartTraceeProcess("TestEventStreamCleanup"))
+            {
+                //TestRunner should account for start delay to make sure that the diagnostic pipe is available.
+
+                var client = new DiagnosticsClient(testExecution.TestRunner.Pid);
+                var settings = new EventTracePipelineSettings()
+                {
+                    Duration = Timeout.InfiniteTimeSpan,
+                    Configuration = new CpuProfileConfiguration()
+                };
+
+                await using var pipeline = new EventTracePipeline(client, settings, (s, token) =>
+                {
+                    eventStream = s; //Clients should not do this.
+                    return Task.CompletedTask;
+                });
+
+                await PipelineTestUtilities.ExecutePipelineWithDebugee(pipeline, testExecution);
```

Comment:
Could you add cancellation to this so we can verify that the stream is disposed due to cancellation? This is currently using the StopAsync method which the test above already verifies.